tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -1791,25 +1785,22 @@ class MklQuantizedConv2DSumReluOp
       MklDnnShape summand_mkl_shape;
       GetMklShape(context, summand_idx, &summand_mkl_shape);
       auto dst_md = summand_mkl_shape.GetMklLayout();
-      if (summand_mkl_shape.IsMklTensor()) {
-        if (summand_type == DT_QINT8) {
-          OP_REQUIRES_OK(context, summand.BitcastFrom(summand, DT_QUINT8,
-                                                      summand.shape()));
-          dst_md.data.data_type =
-              static_cast<mkldnn_data_type_t>(MklDnnType<Toutput>());
-          summand_mkl_shape.SetMklLayout(&dst_md);
-          summand_mkl_shape.SetElemType(MklDnnType<Toutput>());
-        }
-        ForwardMklTensorInToOutWithMklShape(context, summand_idx, 0,
-                                            summand_mkl_shape);
-        *output_tensor = const_cast<Tensor*>(&summand);
-        return;
-      } else {
-        TF_CHECK_OK(Status(error::Code::FAILED_PRECONDITION,
-                           "Current fusion is not successful."));
+
+      // TODO(mdfaijul): handle both non-MKL and MKL tensors
+      if (summand_type == DT_QINT8) {
```

Comment:
So the bug fix is removing the check if `summand` is an MKL tensor? Are the following code okay when `summand` is a TF tensor?

Reply:
Yes. 
Actually the fixed code has been working since Q3 2018, in a private branch (Intel) for all our INT8 work.

Reply:
The private branch is called "int8-master". However, when people of my team submitted public PR's, they mostly did from other private branches.  That is why this fix was never included.

Reply:
Thanks for your clarifications!